factory/runType/sovereignRunTypeCoreComponents.go

```diff
@@ -0,0 +1,31 @@
+package runType
+
+import (
+	"github.com/multiversx/mx-chain-core-go/core/check"
+
+	"github.com/multiversx/mx-chain-go/errors"
+	"github.com/multiversx/mx-chain-go/process/rating"
+	"github.com/multiversx/mx-chain-go/sharding"
+)
+
+type sovereignRunTypeCoreComponentsFactory struct {
```

Comment:
We need to have `IsInterfaceNil` as method mandatory. Inheritance is not the way to go with that